plugins/destination/postgresql/cloud-config-ui/src/form/index.tsx

```diff
@@ -0,0 +1,263 @@
+            render={({ field, fieldState }) => (
+              <TextField
+                error={!!fieldState.error}
+                fullWidth={true}
+                required={true}
+                helperText={
+                  fieldState.error?.message ||
+                  'Maximum size of items that may be grouped together to be written in a single write.'
+                }
+                label="Batch size (bytes)"
+                {...field}
+              />
+            )}
+          />
+          <Controller
+            control={control}
+            name="spec.batchTimeout"
+            render={({ field, fieldState }) => (
+              <TextField
+                error={!!fieldState.error}
+                fullWidth={true}
+                required={true}
+                helperText={fieldState.error?.message || 'Maximum interval between batch writes.'}
+                label="Batch timeout"
+                {...field}
+              />
+            )}
+          />
+          <Controller
+            control={control}
+            name="migrateMode"
+            render={({ field, fieldState }) => (
+              <TextField
+                error={!!fieldState.error}
+                fullWidth={true}
+                required={true}
+                helperText={
+                  fieldState.error?.message || (
+                    <>
+                      Specifies the migration mode to use when source tables are changed.{' '}
+                      <a
+                        href="https://docs.cloudquery.io/docs/reference/destination-spec#migrate_mode"
+                        target="_blank"
+                        rel="noreferrer"
+                      >
+                        Learn more
+                      </a>
+                    </>
+                  )
+                }
+                label="Migrate mode"
+                select={true}
+                SelectProps={{
+                  MenuProps: {
+                    autoFocus: false,
+                    disableAutoFocus: true,
+                  },
+                }}
+                {...field}
+              >
+                {migrateModeValues.map((value) => (
+                  <MenuItem key={value} value={value}>
+                    {value}
+                  </MenuItem>
+                ))}
+              </TextField>
+            )}
+          />
+          <Controller
+            control={control}
+            name="writeMode"
+            render={({ field, fieldState }) => (
+              <TextField
+                error={!!fieldState.error}
+                fullWidth={true}
+                required={true}
+                helperText={
+                  fieldState.error?.message || (
+                    <>
+                      Specifies the update method to use when inserting rows.{' '}
+                      <a
+                        href="https://docs.cloudquery.io/docs/reference/destination-spec#write_mode"
```

Comment:
does this work with the iframe embedded in the main app? Or do we need to use the messenger open_url here?

Reply:
Nice catch, thanks!